main/lsp/tools/make_lsp_types.cc

```diff
@@ -61,6 +61,12 @@ void makeLSPTypes(vector<shared_ptr<JSONClassType>> &enumTypes, vector<shared_pt
     shared_ptr<JSONType> JSONInt = make_shared<JSONIntType>();
     shared_ptr<JSONType> JSONDouble = make_shared<JSONDoubleType>();
     shared_ptr<JSONType> JSONString = make_shared<JSONStringType>();
+    // In the LSP specification JSONAny defined as
+    // `export type LSPAny = LSPObject | LSPArray | string | integer | uinteger | decimal | boolean | null;`
+    // https://microsoft.github.io/language-server-protocol/specifications/specification-3-17/#number
+    //
+    // We don't support LSPObject and LSPArray options yet
```

Comment:
This seems bad. Doesn't this mean that i'll fail at runtime if the client sends back an object? That seems like a pretty common case, doesn't it?

Reply:
Yes, that's correct. But we control the client, which invokes the commands. The extension also controls which types passed as parameters. So the crashes could potentially happen, but in practice that should never happen.

Proper solution would be to redo the `makeVariant` to support multiple non-trivial types. However, I don't know how to do that right now, it may require some head scratching. Potentially that could require remaking how do we generate LSP types in general.